### PR TITLE
IPS & ood-portal-generator: Change whitelist to allowlist

### PR DESCRIPTION
**Pull request by matthu017**
Deals with ips and ood_portal_generator

### CODE REVIEW

File: ood-portal-generator/lib/ood_portal_generator/view.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def initialize(opts = {})
 
       # Maintenance configuration
       @use_maintenance          = opts.fetch(:use_maintenance, true)
-      @maintenance_ip_whitelist = Array(opts.fetch(:maintenance_ip_whitelist, []))
+      @maintenance_ip_allowlist = Array(opts.fetch(:maintenance_ip_allowlist || :maintenance_ip_whitelist, []))
```
Comment (johrstrom) [inline ood-portal-generator/lib/ood_portal_generator/view.rb:33]: I don't think this works with a simple `||`. You'll probably have to break it out into a function.

```
irb(main):009:0> opts = {:bar => "bar"}
=> {:bar=>"bar"}
irb(main):010:0> opts.fetch(:foo || :bar, [])
=> []
```

**Review by johrstrom — APPROVED** [review]
I fixed the merge conflict and rebased to master. I think we're good with this, there's a test case testing the current config and the previous config against the same file, so it's in good shape.

**Comment by johrstrom** [comment]
Can I get a second pair of eyes on this @treydock?

I added test cases that utilize fixture files rather than stubbing `context`. I kinda like testing those boundaries, rather than stubbing classes.  In fact, I'd kinda like to refactor this a little towards that.

**Review by treydock — APPROVED** [review]

**Comment by johrstrom** [comment]
Do you have any thoughts on the testing strategy I'd like to move towards?  Input fixture files are small enough and won't need to change alot - but output fixture files would need changing across potentially _all_ of them.  That's kind of the situation we're in now, but I feel like this library's purpose is to write a file given an input file - so let's test those boundaries.

**Comment by treydock** [comment]
I think it makes to stub files and test whole files since that's how the application is operating, it works off of files and writes out whole files.  The huge benefit to validating entire files is to ensure we know 100% how certain changes affect the entire templates.

**Pull request merged**